### ISSUE TITLE
Add version numbers to compiled dlls

### ISSUE THREAD

**Issue by PhilLab** [opencv/opencv#10203]
To make debugging from crash dump files easier or to organize different OpenCV builds, one needs to know the used dll's version.
The OpenCV dlls do not have the version set, see the following screenshot from a crash dump:

![image](https://user-images.githubusercontent.com/1710255/33431288-f58d28f8-d5d3-11e7-88f3-688101628faa.png)

We did the following for our own software but I could not generalize it for OpenCV without breaking the other platform builds, maybe somebody can help here:

1. Created file version.rc
```
#include <winres.h>
#include "Version.h"
/* Version.h defines the following:
#define VER_VERSION_MAJOR 17
#define VER_VERSION_MINOR 06
#define VER_VERSION_REVISION 003
#define VER_VERSION_STRING "17 / M6c"
*/

VS_VERSION_INFO VERSIONINFO
 FILEVERSION VER_VERSION_RC
 PRODUCTVERSION VER_VERSION_RC
 FILEFLAGSMASK 0x17L
#ifdef _DEBUG
 FILEFLAGS 0x1L
#else
 FILEFLAGS 0x0L
#endif
 FILEOS 0x4L
 FILETYPE 0x1L
 FILESUBTYPE 0x0L
BEGIN
    BLOCK "StringFileInfo"
    BEGIN
        BLOCK "040904b0"
        BEGIN
            VALUE "FileDescription", "our software product description"
            VALUE "ProductName", "out product"
            VALUE "ProductVersion", VER_VERSION_STRING
        END
    END
    BLOCK "VarFileInfo"
    BEGIN
        VALUE "Translation", 0x409, 1200
    END
END
```

2. Add it to the source files in cmake

**Comment by alalek** [comment]
Did you mean this: #8947 ? (it is available since OpenCV 3.3)

**Comment by PhilLab** [comment]
Oh, I overlooked that. Yes, I suppose that's it 👍 Thanks for that!
